internal/injector/builtin/yaml/go-main.yml

```diff
@@ -18,39 +18,55 @@ aspects:
+          # already one in the file, but as it currently is appended AFTER all other declarations in the file, it means
+          # that it will be executed last (tracing contents of previous init functions will not be possible).
           template: |-
-            tracer.Start(tracer.WithOrchestrion(map[string]string{"version": {{printf "%q" Version}}}))
-            defer tracer.Stop()
-
-            switch os.Getenv("DD_PROFILING_ENABLED") {
-            case "1", "true", "auto":
-              // The "auto" value can be set if profiling is enabled via the
-              // Datadog Admission Controller. We always turn on the profiler in
-              // the "auto" case since we only send profiles after at least a
-              // minute, and we assume anything running that long is worth
-              // profiling.
-              err := profiler.Start(
-                profiler.WithProfileTypes(
-                  profiler.CPUProfile,
-                  profiler.HeapProfile,
-                  // Non-default profiles which are highly likely to be useful:
-                  profiler.GoroutineProfile,
-                  profiler.MutexProfile,
-                ),
-                profiler.WithTags("orchestrion:true"),
-              )
-              if err != nil {
-                // TODO: is there a better reporting mechanism?
-                // The tracer and profiler already use the stdlib logger, so
-                // we're not adding anything new. But users might be using a
-                // different logger.
-                log.Printf("failed to start profiling: %s", err)
+            func init() {
+              switch os.Getenv("DD_PROFILING_ENABLED") {
+              case "1", "true", "auto":
+                // The "auto" value can be set if profiling is enabled via the
+                // Datadog Admission Controller. We always turn on the profiler in
+                // the "auto" case since we only send profiles after at least a
+                // minute, and we assume anything running that long is worth
+                // profiling.
+                err := profiler.Start(
+                  profiler.WithProfileTypes(
+                    profiler.CPUProfile,
+                    profiler.HeapProfile,
+                    // Non-default profiles which are highly likely to be useful:
+                    profiler.GoroutineProfile,
+                    profiler.MutexProfile,
+                  ),
+                  profiler.WithTags("orchestrion:true"),
+                )
+                if err != nil {
+                  // TODO: is there a better reporting mechanism?
+                  // The tracer and profiler already use the stdlib logger, so
+                  // we're not adding anything new. But users might be using a
+                  // different logger.
+                  log.Printf("failed to start profiling: %s", err)
+                }
+                defer profiler.Stop()
```

Comment:
This will stop the profiler as soon as `func init()` returns, meaning we won't get profiling. We could either keep the profiler start in `main`, or remove the call to `profiler.Stop`. I think either would work fine.

I still plan to set up some end-to-end testing of Orchestrion + profiling which would catch this, just had other priorities recently.

Reply:
Oh yikes, obviously! Good thing you're noticing this...